src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
```

Comment:
Need to to some refactoring in order to improve code readability
```suggestion
  const children = onlyWithSon
    ? people.filter(p => 
        p.sex === 'm'
        && people.some(m => m.name === p.mother)
      )
    : people.filter(p => 
        people.some(m => m.name === p.mother)
      );

  const sum = children.reduce((acc, { born, mother }) => {
    const founded = people.find(({ name }) => name === mother);
    acc += founded?.born || 0;
    
    return acc;
  }, 0);
```